pkgs/README.md

```diff
@@ -741,6 +741,34 @@ stdenv.mkDerivation {
 }
 ```
 
+### Version Testing
+
+Many programs provide a command-line flag that emits version information. It provides a straightforward test procedure, namely, run the program with that specific flag and verify the output.
+
+Nixpkgs provides two tools for this task: [`testers.testVersion`](#tester-testVersion) and [`versionCheckHook`](#versioncheckhook).
+
+#### Comparison between `versionCheckHook` and `testers.testVersion`
+
+The most notable difference between `versionCheckHook` and `testers.testVersion` is that `versionCheckHook` runs at the build time of the package, while `passthru.tests.versions` executes a whole derivation that depends on the package.
+
+The advantages of `testers.testVersion` over `versionCheckHook` are listed at [Package tests](#var-passthru-tests-packages) already. Below we will list some advantages of `versionCheckHook` over `testers.testVersion`:
```

Comment:
Hmm it felt natural how the advantages of `testVersion` were listed there, but here it doesn't feel natural at all how the advantages of `versionCheckHook` are listed here and not in it's documentation. If this section is a comparison, it should aggregate advantages and disadvantages of both methods in the same place.

Reply:
[You suggested to put the comparison in pkgs/README](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1807456582) before.

Do you think this comparison should be in versionCheckHook.md instead?

Reply:
> [You suggested to put the comparison in pkgs/README](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1807456582) before.

I know, but I didn't anticipate that it'd come out so inconsistent, because you'd keep the pros and cons of `testVersion` elsewhere. What I suggested was (this time more accurately):

Put the 2x2 matrix of pros & cons of both `testVersion` & `versionCheckHook` in a neutral place like `pkgs/README.md`, and link to that comparison section from both `testVersion`'s documentation and `versionCheckHook.section.md`. Mention shortly the motivation to use either tests in `testVersion`'s documentation and `versionCheckHook.section.md`.

I cannot promise I'd approve such a change never the less, because if we can (and I think we should) reach a consensus that one of the approaches should be the default approach to use, there is no justification for putting such an exhaustive 2x2 matrix of pros and cons in a neutral place.

Reply:
`pkgs/readme.md` does not work.
For now I will move this to versioncheckhook docs